client/my-sites/posts/post.jsx

```diff
@@ -279,6 +274,10 @@ const Post = React.createClass( {
 		this.setState( { showShare: ! this.state.showShare } );
 	},
 
+	setCommentsFilter( commentsFilter ) {
```

Comment:
Look at this method again :), it's not expecting an object, and works with anything: `this.setState( { commentsFilter } )` (though hopefully it's receiving a function). Or did I misunderstand?